bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -128,7 +128,13 @@ void ReplicaImp::registerMsgHandlers() {
 template <typename T>
 void ReplicaImp::messageHandler(MessageBase *msg) {
   T *trueTypeObj = new T(msg);
+  validateMessage(trueTypeObj);
+  if (isCollectingState()) {
+    // Extract only required information and handover to ST thread
+    ReplicaForStateTransfer::peekConsensusMessage<T>(msg);
```

Comment:
Why not
`peekConsensusMessage<T>(msg);`

Reply:
Thanks. Addressed.